static/service-worker.js

```diff
@@ -22,24 +21,25 @@ self.addEventListener('activate', (event) => {
 self.addEventListener('fetch', function(event) {
   const accept = event.request.headers.get('Accept');
   if (accept === 'application/json' || event.request.method !== 'GET' ||
-      event.request.mode !== 'no-cors') {
+      !['no-cors', 'navigate'].includes(event.request.mode)) {
     return;
   }
+
   const url = new URL(event.request.url);
-  if (url.origin !== location.origin ||
-      !url.pathname.match(CACHE_PATH)) {
+  if (url.origin !== location.origin || url.pathname.endsWith('.map')) {
     return;
   }
+
   event.respondWith((async () => {
```

Comment:
If return cached `index.html` only when request failed, Following code works for me.
```
  event.respondWith((async () => {
    const cache = await caches.open(CACHE);
    if (event.request.mode !== 'navigate') {
      const matching = await cache.match(event.request);
      if (matching) {
        return matching;
      }
    }

    try {
      const response = await fetch(event.request);
      await cache.put(event.request, response.clone());
      return response;
    } catch (e) {
      return cache.match(event.request or cached 'index.html');
    }
  })());
```